jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -688,7 +688,7 @@ public Response payAllInvoices(@PathParam("accountId") final String accountId,
                                            invoice.getBalance() : remainingRequestPayment;
             if (amountToPay.compareTo(BigDecimal.ZERO) > 0) {
                 final UUID paymentMethodId = externalPayment ? null : account.getPaymentMethodId();
-                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, pluginProperties, callContext);
+                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, UUIDs.randomUUID().toString(), UUIDs.randomUUID().toString(), pluginProperties, callContext);
```

Comment:
@pierre @sbrossie 
So, I removed the precondition and tried (from a test) to make sure that the transaction external key is not null.
You said that should be equal to the **id**, which id did you mean? the transactionId? 🤔
At the end of the process, the transactionExternalKey has always an UUID value but does not match with any other ID value associated. So I debugged for quite a while until I got [this!](https://github.com/javier-gomez-hs/killbill/blob/b31a8ad2e463c5bf9869b4d2c69078e3ea5ecb55/payment/src/main/java/org/killbill/billing/payment/core/sm/control/DefaultControlInitiated.java#L94-L94)
Apparently, if at that time the transactionExternalKey is still null, a **random** value is being set 😒
What is your suggestion on that? 

Please let me know and I can make the changes!

Thanks!